presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -135,6 +143,7 @@ public static CachingHiveMetastore memoizeMetastore(ExtendedHiveMetastore delega
         return new CachingHiveMetastore(
                 delegate,
                 newDirectExecutorService(),
+                true,
```

Comment:
This is incorrect. It must be set to the configuration value. While it currently doesn't effectively affect existing call sites it might at some point if it is used out of a "single query" context.